lib/mongodb.js

```diff
@@ -483,6 +482,121 @@ MongoDB.prototype.updateOrCreate = function updateOrCreate(model, data, options,
 };
 
 /**
+ * Parses the data input for replace operations and returns the
+ * sanitised version of the object.
+ *
+ * @param data
+ * @returns {*}
+ */
```

Comment:
@bajtos: Again probably it makes more sense to have an argument in `parseUpdateData` for `replace` and if value `true` is passed, we unset the missing fields by `parsedData.$unset[key] = "";`